src/tlo/methods/cervical_cancer.py

```diff
@@ -0,0 +1,1542 @@
+        self.EXPECTED_APPT_FOOTPRINT = self.make_appt_footprint({"Over5OPD": 1})
+        self.ACCEPTED_FACILITY_LEVEL = '1a'
+
+    def apply(self, person_id, squeeze_factor):
+        df = self.sim.population.props
+        person = df.loc[person_id]
+        hs = self.sim.modules["HealthSystem"]
+
+        # Run a test to diagnose whether the person has condition:
+        dx_result = hs.dx_manager.run_dx_test(
+            dx_tests_to_run='screening_with_via_for_cin_and_cervical_cancer',
+            hsi_event=self
+        )
+
+        if dx_result:
+            df.at[person_id, 'ce_via_cin_ever_detected'] = True
+
+        if dx_result and (df.at[person_id, 'ce_hpv_cc_status'] == 'cin1'
+                        or df.at[person_id, 'ce_hpv_cc_status'] == 'cin2'
+                        or df.at[person_id, 'ce_hpv_cc_status'] == 'cin3'
+                        ):
+            hs.schedule_hsi_event(
+                    hsi_event=HSI_CervicalCancer_Cryotherapy_CIN(
+                        module=self.module,
+                        person_id=person_id
+                           ),
+                    priority=0,
+                    topen=self.sim.date,
+                    tclose=None
+                           )
+
+        if dx_result and (df.at[person_id, 'ce_hpv_cc_status'] == 'stage1'
+                        or df.at[person_id, 'ce_hpv_cc_status'] == 'stage2a'
+                        or df.at[person_id, 'ce_hpv_cc_status'] == 'stage2b'
+                        or df.at[person_id, 'ce_hpv_cc_status'] == 'stage3'
+                        or df.at[person_id, 'ce_hpv_cc_status'] == 'stage4'):
+            hs.schedule_hsi_event(
+                hsi_event=HSI_CervicalCancer_Biopsy(
+                    module=self.module,
+                    person_id=person_id
+                ),
+                priority=0,
+                topen=self.sim.date,
+                tclose=None
+            )
+
+        # sy_chosen_via_screening_for_cin_cervical_cancer reset to 0
+        if df.at[person_id, 'sy_chosen_via_screening_for_cin_cervical_cancer'] == 2:
+            self.sim.modules['SymptomManager'].change_symptom(
+                person_id=person_id,
+                symptom_string='chosen_via_screening_for_cin_cervical_cancer',
+                add_or_remove='-',
+                disease_module=self.module
+                )
+
+        df.at[person_id, 'ce_selected_for_via_this_month'] = False
+
+
+class HSI_CervicalCancer_XpertHPVScreening(HSI_Event, IndividualScopeEventMixin):
+
+    """
+     This event will be scheduled by family planning HSI - for now we determine at random a screening event, and
+     we determine at random whether this is AceticAcidScreening or HPVXpertScreening
+
+     In future this might be scheduled by the contraception module
+    """
+
+    def __init__(self, module, person_id):
+        super().__init__(module, person_id=person_id)
+
+        self.TREATMENT_ID = "CervicalCancer_XpertHPVScreening"
```

Comment:
I think some equipment declaration needed for this (and others?) HSI_Events?

Reply:
yes I will be talking to Joe about that